### PR TITLE
refactor: change shape of `Validation.Success`

### PR DESCRIPTION
**Pull request by jaschdoc**
- [ ] Add tests

Related to https://github.com/flix/flix/issues/5026

### CODE REVIEW

**Comment by magnus-madsen** [comment]
I think we will want some new tests too.

**Comment by jaschdoc** [comment]
Community build fails because of https://github.com/flix/flix/issues/5112

**Comment by jaschdoc** [comment]
> I think we will want some new tests too.

Might be a good idea yeah hehe

File: main/test/ca/uwaterloo/flix/language/TestFlixErrors.scala [hunk 1]
```diff
@@ -26,7 +26,7 @@ class TestFlixErrors extends FunSuite with TestUtils {
   def expectRuntimeError(v: Validation[CompilationResult, CompilationMessage], name: String): Unit = {
     expectSuccess(v)
     v match {
-      case Validation.Success(t) => t.getMain match {
+      case Validation.Success(t, _) => t.getMain match {
```
Comment (jaschdoc) [inline main/test/ca/uwaterloo/flix/language/TestFlixErrors.scala:29]: Should this be changed to `Validation.Success(t, e) if e.isEmpty => ...`? and then handle non-empty errors like failures?

File: main/test/ca/uwaterloo/flix/language/TestProgramArgs.scala [hunk 2]
```diff
@@ -35,7 +35,7 @@ class TestProgramArgs extends FunSuite with TestUtils {
       """.stripMargin
     val result = compile(input, Options.TestWithLibAll)
     result match {
-      case Validation.Success(result) => result.getMain match {
+      case Validation.Success(result, _) => result.getMain match {
```
Comment (jaschdoc) [inline main/test/ca/uwaterloo/flix/language/TestProgramArgs.scala:38]: Same as above

File: main/test/ca/uwaterloo/flix/util/FlixSuite.scala [hunk 3]
```diff
@@ -63,7 +63,7 @@ class FlixSuite(incremental: Boolean) extends FunSuite {
     try {
       // Compile and Evaluate the program to obtain the compilationResult.
       flix.compile() match {
-        case Success(compilationResult) =>
+        case Success(compilationResult, _) =>
```
Comment (jaschdoc) [inline main/test/ca/uwaterloo/flix/util/FlixSuite.scala:66]: Same as above

File: main/test/ca/uwaterloo/flix/TestUtils.scala [hunk 4]
```diff
@@ -43,7 +43,7 @@ trait TestUtils {
     * Asserts that the validation is a failure with a value of the parametric type `T`.
     */
   def expectError[T](result: Validation[CompilationResult, CompilationMessage])(implicit classTag: ClassTag[T]): Unit = result match {
-    case Validation.Success(_) => fail(s"Expected Failure, but got Success.")
+    case Validation.Success(_, _) => fail(s"Expected Failure, but got Success.")
```
Comment (jaschdoc) [inline main/test/ca/uwaterloo/flix/TestUtils.scala:46]: Same as other comments

**Comment by mlutze** [comment]
> Community build fails because of #5112

Everyone yell at @JonathanStarup until he released a new version

**Comment by JonathanStarup** [comment]
> > Community build fails because of #5112
> 
> Everyone yell at @JonathanStarup until he releases a new version

oooh yeah I forgot about the release part 😄 I just committed and went my merry way

File: main/test/ca/uwaterloo/flix/language/TestFlixErrors.scala (hunk 1 above)
Comment (magnus-madsen) [reply main/test/ca/uwaterloo/flix/language/TestFlixErrors.scala:29]: Yes, this is the main concern. How we can be sure to do this right everywhere.

**Comment by magnus-madsen** [comment]
I think, probably in a new PR, we should try with an extra case: `SuccessWithFailures`.

**Pull request closed without merging**